integration-tests/tests/common.rs

```diff
@@ -0,0 +1,204 @@
+use aws_config;
+use aws_sdk_secretsmanager;
+use std::env;
+use std::path::PathBuf;
+use std::process::Stdio;
+use std::time::Duration;
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::process::Command as TokioCommand;
+use url::Url;
+
+#[derive(Debug)]
+pub struct AgentQuery {
+    pub secret_id: String,
+    pub version_id: Option<String>,
+    pub version_stage: Option<String>,
+    pub refresh_now: Option<bool>,
+}
+
+impl AgentQuery {
+    pub fn new(secret_id: impl Into<String>) -> Self {
+        Self {
+            secret_id: secret_id.into(),
+            version_id: None,
+            version_stage: None,
+            refresh_now: None,
+        }
+    }
+
+    pub fn with_version_id(mut self, version_id: impl Into<String>) -> Self {
```

Comment:
There's a macro crate for this https://docs.rs/derive_builder/latest/derive_builder/